app/src/main/java/com/example/goodnote/ui/viewModels/NoteViewModel.kt

```diff
@@ -27,37 +30,34 @@ class NoteViewModel(private val repository: NoteRepo) : ViewModel() {
 
     fun getAllNotes() = viewModelScope.launch {
         val notes = withContext(Dispatchers.IO) { repository.getAllNotes() }
-        _repoNotes.value = notes
+
+        _repoNotes.value = notes.toListNoteListModel()
     }
 
-    fun saveNote(note: Note) = viewModelScope.launch {
+    fun saveNote(note: NoteDetailsModel) = viewModelScope.launch {
 
-        if (note.title.isNullOrEmpty()) note.title = DEFAULT_TITLE
+        val noteSave = if (note.title.isNullOrEmpty()) note.copy(title = DEFAULT_TITLE) else note
 
-        _repoNotes.addOne(note) // --> do not make a new list, but just add a new note (DiffUtil)
+        _repoNotes.addOne(noteSave.toNoteListModel()) // --> do not make a new list, but just add a new note (DiffUtil)
```

Comment:
Implementation of this method is doing exactly what you said you don't want it to do :D
It creates a new list by merging a new object with the old notes list. :)

Please take a closer look at the implementation of this method :)

Reply:
I know, I haven't even touched DiffUtil yet :D

Reply:
DiffUtil never changes position/index of items in new list